rundeckapp/grails-spa/packages/ui-trellis/src/library/components/storage/tests/KeyStoragePage.spec.ts

```diff
@@ -0,0 +1,171 @@
+    // Simulate the cancelEditing event
+    const keyStorageEdit = wrapper.findComponent({ name: "KeyStorageEdit" });
+    await keyStorageEdit.vm.$emit("cancelEditing");
+    await wrapper.vm.$nextTick();
+    await flushPromises();
+    const emittedEvents = keyStorageEdit.emitted().cancelEditing;
+    expect(emittedEvents).toHaveLength(1);
+  });
+  it("emits finishEditing when a key is created or modified", async () => {
+    const wrapper = await mountKeyStoragePage();
+    await wrapper.vm.$nextTick();
+    const keyStorageEdit = wrapper.findComponent({ name: "KeyStorageEdit" });
+    const selectedKey = {
+      name: "newKey",
+      path: "/keys/newKey",
+      keyType: "privateKey",
+      modifyMode: true,
+      fileContent: "file-content",
+      textArea: "some-text",
+      status: "update",
+      password: "password",
+      errorMsg: null,
+    };
+    await keyStorageEdit.vm.$emit("finishEditing", selectedKey);
+    await wrapper.vm.$nextTick();
+    const emittedEvents = keyStorageEdit.emitted().finishEditing;
+    expect(emittedEvents[0]).toEqual([selectedKey]);
+    expect(emittedEvents).toHaveLength(1);
+  });
+
+  it("updates selectedKey when key-created event is emitted", async () => {
+    const wrapper = await mountKeyStoragePage();
+    await wrapper.vm.$nextTick();
+    // 1. First mock: Initial files before any event
+    rundeckClientMock.storageKeyGetMetadata.mockResolvedValueOnce({
+      resources: [
+        {
+          name: "testKey",
+          path: "/keys/testKey",
+          type: "file",
+          meta: { rundeckKeyType: "private" },
+        },
+      ],
+      _response: { status: 200 },
+    });
+    const keyStorageView = wrapper.findComponent({ name: "KeyStorageView" });
+    const newKey = {
+      name: "newKey",
+      path: "/keys/newKey",
+      keyType: "privateKey",
+      meta: { rundeckKeyType: "private" },
+    };
+    await flushPromises();
+    await wrapper.vm.$nextTick();
+    // 2. Second mock: After the  event, reflecting the new key
+    rundeckClientMock.storageKeyGetMetadata.mockResolvedValueOnce({
+      resources: [
+        {
+          name: "newKey",
+          path: "/keys/newKey",
+          type: "file",
+          meta: { rundeckKeyType: "private" },
+        },
+      ],
+      _response: { status: 200 },
+    });
+    const keyStorageEdit = wrapper.findComponent({ name: "KeyStorageEdit" });
+    await keyStorageEdit.vm.$emit("keyCreated", newKey);
+
+    const addKeyButton = wrapper.find('[data-testid="add-key-btn"]');
+    await addKeyButton.trigger("click");
+    await flushPromises();
+    await wrapper.vm.$nextTick();
+    // Ensure that the files array is updated after the event and button click
+    expect(keyStorageView.vm.files.length).toBe(1);
+    expect(keyStorageView.vm.files).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          name: "newKey",
+          path: "/keys/newKey",
+        }),
+      ]),
+    );
```

Comment:
Reminder for the future: Whenever possible, we want to assert things by checking what is currently rendered, and not by the `vm.[property]`. Could replace these expects with the assertion that the data-testid created-key was rendered